Sources/Remote Logging/Crash Logging/PerformanceTracking.swift

```diff
@@ -0,0 +1,53 @@
+/// Defines whether to enable performance tracking, and if so, how to configure it.
+public enum PerformanceTracking {
```

Comment:
Have you considered using `Configuration?` to represent enable and disabled performance tracking? By doing this, most of the new getters in `CrashLoggingDataProvider` can be removed.

Reply:
I thought about it but decided to go for the `enum` because it seemed like a more explicit API. 

The value I see in those getters is to test the translation from a `Configuration` value to the Sentry-specific settings.
Of course, one could say, isn't it easier to run:

```swift
guard let configuration = provider.performanceTrackingConfiguration else { return }

// set each Sentry value with a configuration one
```

But to me it felt like those extra test reduced the chance of errors in translating the configuration to only human error in setting "sentry property `a`" with "configuration property `b`".

Reply:
One way I can think of to remove the need for those getters is to add a method like

```swift
func ifEnabled(_ action: (Configuration) -> Void) {
  guard case .enabled(let configuration) = self else { return }
  action(configuration)
}
```

Then we could use it in `CrashLogging` like

```swift
options.enableAutoPerformanceTracking = self.dataProvider.enableAutoPerformanceTracking
self.dataProvider.performanceTracking.ifEnabled { configuration in
    options.enableNetworkTracking = configuration.trackNetwork
    // ...
}
```

Reply:
The current code looks good to me. I just thought removing most of the new getters would be a good side effect of using an optional property. I don't think we _need_ to do something extra to remove those new getters.